### DISCUSSION TITLE
Debug ESP32-S3 with GDB and Built-in JTAG

### DISCUSSION THREAD

**Discussion by rdlaner** [micropython/micropython#11695] [ESP32]
Hey folks, since the esp32s3 boards support built-in USB and JTAG support, I'm trying to come up with a reliable way to live debug with  GDB, but am not having much luck. Would love to hear from anyone who has been able to use GDB with an esp32s3.

For my attempts, I'm using the Unexpected Maker FeatherS3 board. Here is what I'm doing:
* I build release 1.20.0 manually with: `make BOARD=UM_FEATHERS3`
* Erase the board: `esptool.py --chip esp32s3 --port /dev/cu.usbmodem1101 erase_flash`
* Flash the board with the command provided at the end of the build process:
```
/Users/dustinlane/.espressif/python_env/idf4.4_py3.11_env/bin/python ../../../../CircuitPython/circuitpython/ports/espressif/esp-idf/components/esptool_py/esptool/esptool.py -p /dev/cu.usbmodem1101 -b 460800 --before default_reset --after no_reset --chip esp32s3  write_flash --flash_mode dio --flash_size detect --flash_freq 80m 0x0 build-UM_FEATHERS3/bootloader/bootloader.bin 0x8000 build-UM_FEATHERS3/partition_table/partition-table.bin 0x10000 build-UM_FEATHERS3/micropython.bin
```
* Start `openocd` in one shell: `openocd -f ~/.espressif/tools/openocd-esp32/v0.11.0-esp32-20221026/openocd-esp32/share/openocd/scripts/board/esp32s3-builtin.cfg`
* Launch gdb in another shell: `xtensa-esp32s3-elf-gdb build-UM_FEATHERS3/micropython.elf`
* Connect to target, set breakpoint and attempt to run to main:
```
target remote :3333
set remote hardware-watchpoint-limit 2
mon reset halt
flushregs
thb app_main
c
```

At which point, I never make it to my breakpoint. When halting, I try getting a `bt`, but it doesn't appear to be finding the debug symbols or at least is not able to find the source references:
```
Program received signal SIGINT, Interrupt.
0x40041a76 in ?? ()
(gdb) bt
#0  0x40041a76 in ?? ()
#1  0x40047b51 in ?? ()
#2  0x400439a7 in ?? ()
#3  0x40043c36 in ?? ()
#4  0x40034c48 in ?? ()
```

Any idea what I may be missing?

**Comment by xlla** [comment]
I think you should build elf in debug mode.

**Comment by n1zzo** [comment]
I think you're stuck in ESP32 bootloader mode, try to reset the target with the reset button/pin and let it boot, then attach the JTAG.
